parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -115,175 +115,160 @@ propagate patch b = case validatePatch patch of
-                               newTerms
-                               typeEdits'
-                               typeReplacements'
-                               newTypes'
-                     , seen')
-            doTerm :: Reference -> F m i v (Maybe (Edits v), Set Reference)
-            doTerm r = do
-              componentMap <- unhashTermComponent r
-              let componentMap' =
-                    over
-                        _2
-                        (Term.updateDependencies termReplacements
-                                                 typeReplacements
-                        )
-                      <$> componentMap
-                  seen' = seen <> Set.fromList (view _1 <$> Map.elems componentMap)
-              mayComponent <- verifyTermComponent componentMap' es
-              case mayComponent of
-                Nothing -> pure (Nothing, seen')
-                Just componentMap'' -> do
-                  let
-                    joinedStuff = toList
-                      (Map.intersectionWith f componentMap componentMap'')
-                    f (oldRef, _oldTerm, oldType) (newRef, newTerm, newType) =
-                      (oldRef, newRef, newTerm, oldType, newType')
-                        -- Don't replace the type if it hasn't changed.
-                        where newType'
-                               | Typechecker.isEqual oldType newType = oldType
-                               | otherwise = newType
-                -- collect the hashedComponents into edits/replacements/newterms/seen
-                    termEdits' =
-                      termEdits <> (Map.fromList . fmap toEdit) joinedStuff
-                    toEdit (r, r', _newTerm, oldType, newType) =
-                      (r, TermEdit.Replace r' $ TermEdit.typing newType oldType)
-                    termReplacements' = termReplacements
-                      <> (Map.fromList . fmap toReplacement) joinedStuff
-                    toReplacement (r, r', _, _, _) = (r, r')
-                    newTerms' =
-                      newTerms <> (Map.fromList . fmap toNewTerm) joinedStuff
-                    toNewTerm (_, r', tm, _, tp) = (r', (tm, tp))
-                    writeTerms =
-                      traverse_ (\(Reference.DerivedId id, (tm, tp)) -> eval $ PutTerm id tm tp)
-                  writeTerms [ (r, (tm,ty)) | (_old,r,tm,_oldTy,ty) <- joinedStuff ]
-                  pure
-                    ( Just $ Edits termEdits'
-                                   termReplacements'
-                                   newTerms'
-                                   typeEdits
-                                   typeReplacements
-                                   newTypes
-                    , seen'
+    order <- sortDependentsGraph initialDirty
+    let
+      getOrdered :: Set Reference -> Map Int Reference
+      getOrdered rs =
+        Map.fromList [ (i, r) | r <- toList rs, Just i <- [Map.lookup r order] ]
```

Comment:
@runarorama is this the code that ends up filtering the updates to just this branch? Can you explain why it works? As I was reading it wasn't clear to me that `getOrdered` will only return a submap of the references which are found in the current branch.

Reply:
Looks like the filtering actually happens here:

https://github.com/unisonweb/unison/blob/e62a1809fe10889573419eb3e2817d3edc290eb0/parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs#L401

Reply:
So I think what ends up happening is that the patch is populated with all the dependents in the codebase, then the patch gets applied to the current branch.

Reply:
Since this behaviour is a little weird, I've made a change so that they patch only considers the current branch.